### PR TITLE
Large files are put using icommands. Some refactoring.

### PR DESCRIPTION
**Pull request by kxk302**
(Please replace this header with a description of your pull request. Please include *BOTH* what you did and why you made the changes. The "why" may simply be citing a relevant Galaxy issue.)
(If fixing a bug, please add any relevant error or traceback)
(For UI components, it is recommended to include screenshots or screencasts)

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [ ] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these contributions under [Galaxy's current license](https://github.com/galaxyproject/galaxy/blob/dev/LICENSE.txt).
- [x] I agree to allow the Galaxy committers to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT). If this condition is an issue, uncheck and just let us know why with an e-mail to galaxy-committers@lists.galaxyproject.org.

### CODE REVIEW

File: lib/galaxy/config/sample/object_store_conf.xml.sample [hunk 1]
```diff
@@ -52,6 +52,7 @@
             <cache path="database/object_store_cache_irods" size="1000" />
             <extra_dir type="job_work" path="database/job_working_directory_irods"/>
             <extra_dir type="temp" path="database/tmp_irods"/>
+            <icommands put_cutoff="100000000" num_threads="1"/>
```
Comment (kxk302) [inline lib/galaxy/config/sample/object_store_conf.xml.sample:55]: Files larger than put_cutoff are pushed to irods server via icommands. num_threads is 1 for now. But I want to have the flexibility to change it if need be.

File: lib/galaxy/objectstore/irods.py [hunk 2]
```diff
@@ -25,6 +26,16 @@
 IRODS_IMPORT_MESSAGE = ('The Python irods package is required to use this feature, please install it')
 # 1 MB
 CHUNK_SIZE = 2**20
+# 100 MB. Default file size (configurable) after which icommands are used instead of client code for performance reasons
```
Comment (kxk302) [inline lib/galaxy/objectstore/irods.py:29]: Default value of new parameters.

File: lib/galaxy/objectstore/irods.py [hunk 3]
```diff
@@ -25,6 +26,16 @@
...
+ICOMMANDS_PUT_CUTOFF = 100 * 1024 * 1024
+# Default number of threads (configurable) used in icommands for performance reasons
+ICOMMANDS_NUM_THREADS = 1
+
+# irods connection defaults
```
Comment (kxk302) [inline lib/galaxy/objectstore/irods.py:34]: Defined these in place of some magic numbers in code.

File: lib/galaxy/objectstore/irods.py [hunk 4]
```diff
@@ -60,9 +71,9 @@ def parse_config_xml(config_xml):
         if not c_xml:
             _config_xml_error('connection')
         host = c_xml[0].get('host', None)
-        port = int(c_xml[0].get('port', 0))
-        timeout = int(c_xml[0].get('timeout', 30))
-        refresh_time = int(c_xml[0].get('refresh_time', 300))
+        port = int(c_xml[0].get('port', CONNECTION_PORT))
+        timeout = int(c_xml[0].get('timeout', CONNECTION_TIMEOUT))
```
Comment (kxk302) [inline lib/galaxy/objectstore/irods.py:75]: Removed magic numbers

**Comment by hexylena** [comment]
@kxk302 could you please fill out the PR templates, replacing the bits where it asks for description of the fix? It'll help us with release notes.

**Comment by kxk302** [comment]
@hexylena, this should have been opened in draft and now can be closed.Sorry for the confusion.

**Comment by hexylena** [comment]
no worries! just want to make sure everyone fills the template properly to make our lives easier later :)

**Pull request closed without merging**